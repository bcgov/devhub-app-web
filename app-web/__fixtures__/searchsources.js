/*
Copyright 2019 Province of British Columbia

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at 

   http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

Created by Patrick Simonian
*/
// run time apollo query fixtures

export const ROCKET_CHAT = [
  {
    id: '1', 
    roomId: '2', author: 'apple', time: '2019-02-12', message: 'apples are clearly superior',
  },
  {
    id: '2', 
    roomId: '2', author: 'orange', time: '2019-02-12', message: 'Nice joke apple, are trying to perform some time of standup?',
  },
  {
    id: '3', 
    roomId: '2', author: 'apple', time: '2019-02-12', message: 'Don\'t play coy with me orange, you know I\'m telling the truth',
  },
  {
    id: '4', 
    roomId: '2', author: 'orange', time: '2019-02-12', message: 'I ain\'t never hurd of people curing scurvy with apples yo. Hook line and sinker dawg.',
  },
]