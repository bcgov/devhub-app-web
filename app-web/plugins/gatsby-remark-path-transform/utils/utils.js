/*
Copyright 2018 Province of British Columbia

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at 

   http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

Created by Patrick Simonian
*/

const path = require('path');
const jsdom = require('jsdom'); // eslint-disable-line

const { JSDOM } = jsdom;

/**
 * based on the files location within the repo
 * the relative asset path declared in the file
 * and the repo
 * this will return an absolute location of the asset
 * path
 * @param {String} filePath 
 * @param {String} repoURI 
 * @param {String} relativePath 
 * @returns {String} the absolute version of the relative asset path
 */
const assetRelativePathToAbsolute = (filePath, repoURI, relativePath) => {
    // get directory path from filePath
    try {
        const pathToFile = path.parse(filePath).dir;
        const absoluteAssetPath = repoURI + path.resolve(`${pathToFile}`, relativePath);
        return absoluteAssetPath;
    } catch(e) {
        return relativePath;  
    }
    
};

const isRelativePath = path => /^\.\.?\//.test(path);

/**
 * sifts over transformed markdown content (which should be html by now)
 * and replace any relative asset links in hrefs or src attributes with
 * absolute versions
 * @param {String} filePath 
 * @param {String} repoURI 
 * @param {String} content 
 */
const convertRelativePathsToAbsolute = (filePath, repoURI, content) => {
    // jsdomify content
    const dom = new JSDOM(content);
    const { window: { document } } = dom;
    const anchors = document.getElementsByTagName('a');
    const images = document.getElementsByTagName('img');
    const videos = document.getElementsByTagName('video');

    const assetTags = Array.from(anchors)
        .concat(Array.from(images))
        .concat(Array.from(videos));
    // loop over asset tags and relace hrefs or srcs if they are relative
    assetTags.forEach(assetTag => {
        const attributeType = assetTag.tagName.toLocaleLowerCase() === 'a' ? 'href' : 'src'; 
        // does assetTag have a src or href attribute?
        const assetPath = assetTag.getAttribute(attributeType);
        if(isRelativePath(assetPath)) {
            const absoluteAssetPath = assetRelativePathToAbsolute(filePath, repoURI, assetPath);
            assetTag.setAttribute(attributeType, absoluteAssetPath);
        }
    });
    return document.getElementsByTagName('body')[0].innerHTML;
};

module.exports = {
    assetRelativePathToAbsolute,
    isRelativePath,
    convertRelativePathsToAbsolute,
};