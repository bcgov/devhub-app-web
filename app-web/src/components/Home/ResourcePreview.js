/*
Copyright 2019 Province of British Columbia

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at 

   http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

Created by Patrick Simonian
*/
import React from 'react';
import PropTypes from 'prop-types';
import styled from '@emotion/styled';

import { EMOTION_BOOTSTRAP_BREAKPOINTS } from '../../constants/designTokens';
import { ChevronLink } from '../UI/Link';
import { Container, Title, StyledLink, LinkContainer } from './index';
import Card from '../Cards/Card/Card';

export const CardWrapper = styled.div`
  margin: 6px 9px;
`;

export const ResourceContainer = styled.div`
  padding: 15px 0;
  display: flex;
  flex-flow: row wrap;
  justify-content: center;
  ${EMOTION_BOOTSTRAP_BREAKPOINTS.sm} {
    justify-content: flex-start;
  }
`;

// used by react-testing-library dom querying
export const TEST_IDS = {
  container: 'resource-preview-container',
};

// this is a wrapper component that encapsulates cards for topics or other sizes
export const ResourcePreview = ({ title, link, resources }) => (
  <Container data-testid={TEST_IDS.container}>
    <Title>
      <StyledLink to={link.to}>{title}</StyledLink>
    </Title>
    <ResourceContainer>
      {resources.slice(0, 6).map(r => (
        
        <CardWrapper key={r.id}>
          <Card
            type={r.fields.resourceType}
            title={r.fields.title}
            description={r.fields.description}
            image={r.fields.image}
            link={r.fields.pagePaths[0]}
            event={r}
          />
        </CardWrapper>
      ))}
    </ResourceContainer>
    <LinkContainer>
      <ChevronLink to={link.to}>{link.text}</ChevronLink>
    </LinkContainer>
  </Container>
);

ResourcePreview.propTypes = {
  title: PropTypes.string.isRequired,
  link: PropTypes.shape({
    text: PropTypes.string.isRequired,
    to: PropTypes.string.isRequired,
  }),
  resources: PropTypes.arrayOf(PropTypes.object).isRequired,
};

export default ResourcePreview;
