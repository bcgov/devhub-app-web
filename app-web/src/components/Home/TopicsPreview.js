/*
Copyright 2019 Province of British Columbia
Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at 
   http://www.apache.org/licenses/LICENSE-2.0
Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
Created by Patrick Simonian
*/
/**
 * contains the filter menu, and searchbar/cards container
 */
import React, { useMemo } from 'react';

import {
  getFirstNonExternalResource,
  buildFeaturedTopic,
  buildPopularTopic,
} from '../../utils/helpers';
import Topic from '../Topic';

import { ChevronLink } from '../UI/Link';
import { Container as PreviewContainer, Title, StyledLink, LinkContainer } from './index';
import { graphql, useStaticQuery } from 'gatsby';
import { flattenGatsbyGraphQL } from '../../utils/dataHelpers';
import {
  FEATURE_TOPIC_CONFIGURATION,
  DYNAMIC_TOPIC_PATHS,
  FEATURED_CONTENT,
  POPULAR_TOPIC_CONFIGURATION,
} from '../../constants/ui';
import { useDevhubSiphonAndGithubRawNodes } from '../../utils/hooks';

// used for @testing-library/react dom querying
export const TEST_IDS = {
  container: 'topics-container',
};

const TopicContent = topics =>
  topics
    .filter(topic => topic.connectsWith.length > 0)
    .slice(0, 4)
    .map(topic => {
      // resources are grouped by type, 'ungroup' them so we can find the first available
      // non external link to use as the entry page for the topic card
      const allResources = topic.connectsWith;

      return (
        <Topic
          key={topic.id}
          title={topic.name}
          description={topic.description}
          link={{ to: getFirstNonExternalResource(allResources), text: 'View' }}
          resources={allResources}
        />
      );
    });

export const TopicsPreview = () => {
  const { topics } = useStaticQuery(graphql`
    query {
      topics: allTopicRegistryJson {
        edges {
          node {
            id
            name
            description
            connectsWith {
              ...DevhubNodeConnection
            }
          }
        }
      }
    }
  `);

  const [devhubSiphon, githubRaw] = useDevhubSiphonAndGithubRawNodes();

  const popularTopic = buildPopularTopic(
    githubRaw,
    POPULAR_TOPIC_CONFIGURATION.name,
    POPULAR_TOPIC_CONFIGURATION.description,
    DYNAMIC_TOPIC_PATHS.popular,
    POPULAR_TOPIC_CONFIGURATION.minPageViews,
    POPULAR_TOPIC_CONFIGURATION.maxNodes,
  );

  const featuredTopic = buildFeaturedTopic(
    githubRaw.concat(devhubSiphon),
    FEATURE_TOPIC_CONFIGURATION.name,
    FEATURE_TOPIC_CONFIGURATION.description,
    DYNAMIC_TOPIC_PATHS.featured,
    FEATURED_CONTENT,
  );
  const allTopics = [popularTopic, featuredTopic].concat(topics.edges);
  const topicNodes = useMemo(() => flattenGatsbyGraphQL(allTopics), [allTopics]);
  const link = { to: '/topics', text: 'Topics' };

  return (
    <PreviewContainer data-testid={TEST_IDS.container}>
      <Title>
        <StyledLink to={link.to}>Topics</StyledLink>
      </Title>

      {TopicContent(topicNodes)}

      <LinkContainer>
        <ChevronLink to={link.to}>{link.text}</ChevronLink>
      </LinkContainer>
    </PreviewContainer>
  );
};

export default TopicsPreview;
