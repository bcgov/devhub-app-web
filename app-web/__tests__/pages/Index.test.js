import React from 'react';
import { render, cleanup } from '@testing-library/react';
import queryString from 'query-string';
import { ThemeProvider } from 'emotion-theming';
import theme from '../../theme';
import { useStaticQuery } from 'gatsby';
import { Index, TEST_IDS } from '../../src/pages/index';
import {
  SIPHON_NODES,
  TOPICS,
  EVENTS,
  MEETUP_NODES,
  FEATURED_TOPIC,
  JOURNEYS,
  POPULAR_TOPIC,
} from '../../__fixtures__/nodes';

import { useSearch, useSearchGate, useDevhubSiphonAndGithubRawNodes } from '../../src/utils/hooks';
import { TEST_IDS as TOPIC_TEST_IDS } from '../../src/components/Home/TopicsPreview';
import { TEST_IDS as RESOURCE_PREVIEW_TEST_IDS } from '../../src/components/Home/ResourcePreview';
import {
  getFirstNonExternalResource,
  getTextAndLink,
  removeUnwantedResults,
  buildFeaturedTopic,
  buildPopularTopic,
  reduceJourneyToSubwayLine,
} from '../../src/utils/helpers';
import { GITHUB_RAW_NODES } from '../../__fixtures__/nodes';
import { client } from '../../wrapWithProvider';
import { ApolloProvider } from 'react-apollo';
import { ROCKET_CHAT, GITHUB, DOCUMIZE } from '../../__fixtures__/searchsources';

// this adds custom jest matchers from jest-dom
jest.mock('query-string');
// mock out layout
jest.mock('../../src/hoc/Layout.js', () => ({ children }) => children);
// mock out search hook

jest.mock('@reach/router', () => ({
  Location: ({ children }) => children({ location: { pathname: '/', search: '' } }),
}));

jest.mock('../../src/utils/hooks.js');

jest.mock('../../src/utils/helpers.js');

getFirstNonExternalResource.mockReturnValue('foo');
getTextAndLink.mockReturnValue({ to: '/documentation?q=mobile', text: 'two results found' });
useSearchGate.mockReturnValue({ results: [], loading: false });
// jwt time stamps are in seconds. dividing by 1000 to convert date.now ms to s

buildFeaturedTopic.mockReturnValue({ node: FEATURED_TOPIC });
buildPopularTopic.mockReturnValue({ node: POPULAR_TOPIC });
reduceJourneyToSubwayLine.mockReturnValue([{ name: 'foo' }]);

describe('Home Page', () => {
  // when you use graphql to load data into the component
  // all edges are an object of { node: [graphql object]}
  // the topics fixture is the true data without this extra object field
  // so we map it to resemble what graphql would do when passing the data attribute into
  // this component
  const nodes = SIPHON_NODES.map(c => ({ node: c }));
  const topics = TOPICS.map(c => ({ node: c }));
  const events = EVENTS.map(c => ({ node: c }));
  const meetups = MEETUP_NODES.map(c => ({ node: c }));
  const githubRaw = GITHUB_RAW_NODES.map(c => ({ node: c }));
  const journeys = JOURNEYS.map(c => ({ node: c }));
  useDevhubSiphonAndGithubRawNodes.mockReturnValue([nodes, githubRaw]);
  useStaticQuery.mockReturnValue({ topics: { edges: topics } });

  const props = {
    client: {},
    data: {
      allJourneyRegistryJson: {
        edges: journeys,
      },
      allTopicRegistryJson: {
        edges: topics,
      },
      allDevhubSiphon: {
        edges: nodes,
      },
      allGithubRaw: {
        edges: githubRaw,
      },
      allEventbriteEvents: {
        edges: events,
      },
      allMeetupGroup: {
        edges: meetups,
      },
      allMarkdownRemark: {
        edges: githubRaw,
      },
    },
    location: {
      search: '',
    },
  };

  afterEach(() => {
    cleanup();
  });

  test('it matches snapshot, when there is a search and no results an alert box shows up', () => {
    queryString.parse.mockReturnValue({});
    const { container, rerender, queryByTestId } = render(
      <ThemeProvider theme={theme}>
        <ApolloProvider client={client}>
          <Index {...props} />
        </ApolloProvider>
      </ThemeProvider>,
    );
    expect(container.firstChild).toMatchSnapshot();
    let Alert;
    Alert = queryByTestId(TEST_IDS.alert);
    // initially there are valid results and so there should no be no alert
    expect(Alert).not.toBeInTheDocument();
    // mock that query string actually finds search values in the url
    queryString.parse.mockReturnValue({ q: 'foo' });
    // rerender stubbing no results for resources
    useSearch.mockReturnValue([]);
    useSearchGate.mockReturnValue({ results: [], loading: false });
    removeUnwantedResults.mockReturnValue([]);
    rerender(
      <ThemeProvider theme={theme}>
        <ApolloProvider client={client}>
          <Index {...props} />
        </ApolloProvider>
      </ThemeProvider>,
    );
    expect(useSearch).toHaveBeenCalled();
    Alert = queryByTestId(TEST_IDS.alert);

    expect(Alert).toBeInTheDocument();
  });

  test('when there is a search and no local results, but external results, no alert box shows up', () => {
    queryString.parse.mockReturnValue({});
    useSearchGate.mockReturnValue({ results: ROCKET_CHAT, loading: false });
    useSearch.mockReturnValue([]);

    const { rerender, queryByTestId } = render(
      <ThemeProvider theme={theme}>
        <ApolloProvider client={client}>
          <Index {...props} />
        </ApolloProvider>
      </ThemeProvider>,
    );

    let Alert;
    Alert = queryByTestId(TEST_IDS.alert);
    // initially there are valid results and so there should no be no alert
    expect(Alert).not.toBeInTheDocument();
    // mock that query string actually finds search values in the url
    queryString.parse.mockReturnValue({ q: 'foo' });
    // rerender stubbing no results for resources
    rerender(
      <ThemeProvider theme={theme}>
        <ApolloProvider client={client}>
          <Index {...props} />
        </ApolloProvider>
      </ThemeProvider>,
    );

    Alert = queryByTestId(TEST_IDS.alert);

    expect(Alert).not.toBeInTheDocument();
  });

  test("When a blank search is entered, cards and alerts don't show but topics do", () => {
    queryString.parse.mockReturnValue({});
    const { container, rerender, queryByTestId, queryAllByTestId } = render(
      <ThemeProvider theme={theme}>
        <ApolloProvider client={client}>
          <Index {...props} />
        </ApolloProvider>
      </ThemeProvider>,
    );
    expect(container.firstChild).toMatchSnapshot();
    let Alert;
    Alert = queryByTestId(TEST_IDS.alert);
    // initially there are valid results and so there should no be no alert
    expect(Alert).not.toBeInTheDocument();
    // mock that query string actually finds search values in the url
    queryString.parse.mockReturnValue({ q: '' });
    // rerender stubbing no results for resources
    useSearch.mockReturnValue([]);
    rerender(
      <ThemeProvider theme={theme}>
        <ApolloProvider client={client}>
          <Index {...props} />
        </ApolloProvider>
      </ThemeProvider>,
    );
    expect(useSearch).toHaveBeenCalled();
    Alert = queryByTestId(TEST_IDS.alert);

    expect(Alert).not.toBeInTheDocument();

    expect(queryByTestId(TOPIC_TEST_IDS.container)).toBeInTheDocument();
    expect(queryAllByTestId(RESOURCE_PREVIEW_TEST_IDS.container).length).toBe(0);
    //The above changed to "toBe(0)" from "toBeGreaterThan(0)" as previews are no longer shown on the home page (unless a valid search has been made)
  });

  test('when searching, topics disappear if there are results', () => {
    queryString.parse.mockReturnValue({ q: 'foo' });
    useSearch.mockReturnValue([{ id: SIPHON_NODES[0].id }]);
    removeUnwantedResults.mockReturnValue([SIPHON_NODES[0]]);
    const { queryByTestId, queryAllByTestId } = render(
      <ThemeProvider theme={theme}>
        <ApolloProvider client={client}>
          <Index {...props} />
        </ApolloProvider>
      </ThemeProvider>,
    );

    expect(queryByTestId(TOPIC_TEST_IDS.container)).not.toBeInTheDocument();
    expect(queryAllByTestId(RESOURCE_PREVIEW_TEST_IDS.container).length).toBeGreaterThan(0);
  });

  test('when there is no search, topics are visible', () => {
    queryString.parse.mockReturnValue({});
    useSearch.mockReturnValue([]);

    const { getByTestId } = render(
      <ThemeProvider theme={theme}>
        <ApolloProvider client={client}>
          <Index {...props} />
        </ApolloProvider>
      </ThemeProvider>,
    );

    expect(getByTestId(TOPIC_TEST_IDS.container)).toBeInTheDocument();
  });

  test('Unwanted results are correctly removed from search results', () => {
    //Our results have siphons node and meetups
    const initialResults = SIPHON_NODES.concat(MEETUP_NODES);
    //in our call to removeUnwantedResults, we say that only events are current (not meetups)
    //thus it should remove the meetups from the initial results
    const filteredResults = removeUnwantedResults(
      initialResults,
      EVENTS.concat(MEETUP_NODES),
      EVENTS,
    );
    expect(filteredResults.length).toBeLessThan(initialResults.length);
  });

  test('shows rocket chat results when authenticated', () => {
    queryString.parse.mockReturnValue({ q: 'foo' });
    useSearch.mockReturnValue([]);
    useSearchGate.mockReturnValue({ results: ROCKET_CHAT, loading: false });

    const { queryByTestId, rerender } = render(
      <ThemeProvider theme={theme}>
        <ApolloProvider client={client}>
          <Index {...props} location={{ search: '?q=foo' }} />
        </ApolloProvider>
      </ThemeProvider>,
    );

    expect(queryByTestId(ROCKET_CHAT[0].id)).toBeInTheDocument();

    rerender();

    expect(queryByTestId(ROCKET_CHAT[0].id)).not.toBeInTheDocument();
  });

  test('shows github results when authenticated', () => {
    queryString.parse.mockReturnValue({ q: 'foo' });
    useSearch.mockReturnValue([]);
    useSearchGate.mockReturnValue({ results: GITHUB, loading: false });
    // useAuthenticated.mockReturnValue(true);

    const { queryByTestId, rerender } = render(
      <ThemeProvider theme={theme}>
        <ApolloProvider client={client}>
          <Index {...props} location={{ search: '?q=foo' }} />
        </ApolloProvider>
      </ThemeProvider>,
    );

    expect(queryByTestId(GITHUB[0].id)).toBeInTheDocument();

    // useAuthenticated.mockReturnValue(false);

    rerender();

    expect(queryByTestId(ROCKET_CHAT[0].id)).not.toBeInTheDocument();
  });

  test('shows documize results in different authenticated status', () => {
    queryString.parse.mockReturnValue({ q: 'openshift' });
    useSearch.mockReturnValue([]);
    useSearchGate.mockReturnValue({ results: DOCUMIZE, loading: false });
    // useAuthenticated.mockReturnValue(true);

    const { queryByTestId, rerender } = render(
      <ThemeProvider theme={theme}>
        <ApolloProvider client={client}>
          <Index {...props} location={{ search: '?q=openshift' }} />
        </ApolloProvider>
      </ThemeProvider>,
    );

    expect(queryByTestId(DOCUMIZE[0].id)).toBeInTheDocument();
    rerender();
    // useAuthenticated.mockReturnValue(false); // do not show when not authenticated.
    expect(queryByTestId(ROCKET_CHAT[0].id)).not.toBeInTheDocument();
    expect(queryByTestId(GITHUB[0].id)).not.toBeInTheDocument();
    expect(queryByTestId(DOCUMIZE[0].id)).not.toBeInTheDocument();
  });
});
