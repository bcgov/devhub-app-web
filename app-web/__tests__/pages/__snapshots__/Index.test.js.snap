// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page When a blank search is entered, cards and alerts don't show but topics do 1`] = `
.emotion-19 {
  background-color: #fafafa;
  text-align: center;
  font-size: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 25px 15px 30px;
  -webkit-flex-flow: column nowrap;
  -ms-flex-flow: column nowrap;
  flex-flow: column nowrap;
}

@media (min-width:576px) {
  .emotion-19 {
    padding: 20px 15px;
  }
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 1em;
}

.emotion-0 {
  margin: 6px 3px 6px 0;
  padding: 0px 4px;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1.54912em;
}

.emotion-2 {
  text-transform: uppercase;
  font-weight: 600;
  color: #fcba19;
}

.emotion-6 {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  color: #1A5A96;
  font-weight: 400;
  text-transform: capitalize;
  padding: 0 4px;
  margin: 0 2px;
}

.emotion-15 {
  width: 100%;
  max-width: 500px;
}

.emotion-11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 1.25em;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
}

.emotion-11 > input {
  margin: 0 4px;
  padding: 4px 5px;
  -webkit-flex-basis: 250px;
  -ms-flex-preferred-size: 250px;
  flex-basis: 250px;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-transition: all 0.25s;
  transition: all 0.25s;
}

.emotion-11 > button {
  padding: 4px 15px;
  margin: 4px;
}

.emotion-11 > button {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.emotion-8 {
  color: #fff;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  border: 1px solid transparent;
  padding: 0.375rem 0.75rem;
  font-size: 1em;
  line-height: 1.5;
  border-radius: 0.25rem;
  cursor: pointer;
  font-weight: 400;
  text-align: center;
  white-space: nowrap;
  vertical-align: middle;
  display: inline-block;
  background-color: #003366;
}

.emotion-13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
}

.emotion-17 {
  display: inline block;
  padding: 15px;
  max-width: 220px;
}

.emotion-17 img {
  width: 100%;
  margin-bottom: 0;
}

<div
  class="emotion-19 emotion-20"
>
  <div
    class="emotion-4 emotion-5"
  >
    <h2
      class="emotion-0 emotion-1"
    >
      DevHub
    </h2>
    <div
      aria-label="This application is currently in beta phase"
      class="emotion-2 emotion-3"
    >
      beta
    </div>
  </div>
  <h4
    css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
  >
    One place that brings together resources to help build digital products for the BC Government
  </h4>
  <a
    class="emotion-6 emotion-7"
    css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
    data-testid="ui-gatsby-link"
    href="/aboutDevhub"
  >
    How Devhub Works
     
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-chevron-right fa-w-10 "
      data-icon="chevron-right"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 320 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z"
        fill="currentColor"
      />
    </svg>
  </a>
  <div
    class="emotion-15 emotion-16"
  >
    <div
      class="emotion-10 emotion-11 emotion-12"
    >
      <input
        aria-label="Search input field"
        class="Input custom-text"
        data-testid="searchbar-input"
        id="dh-search-input"
        maxlength="40"
        type="text"
        value=""
      />
      <button
        aria-label="Click to Search"
        class="emotion-8 emotion-9"
        color="blue"
        data-testid="searchbar-button"
      >
        Search
      </button>
    </div>
    <div
      class="emotion-13 emotion-14"
    />
  </div>
  <div
    class="emotion-17 emotion-18"
    data-testid="Masthead.algolia"
    style="padding-bottom: 5px;"
  >
    <a
      href="https://www.algolia.com/"
    >
      <img
        alt="Algolia Logo"
        src="algolia.svg"
      />
    </a>
  </div>
</div>
`;

exports[`Home Page it matches snapshot, when there is a search and no results an alert box shows up 1`] = `
.emotion-19 {
  background-color: #fafafa;
  text-align: center;
  font-size: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 25px 15px 30px;
  -webkit-flex-flow: column nowrap;
  -ms-flex-flow: column nowrap;
  flex-flow: column nowrap;
}

@media (min-width:576px) {
  .emotion-19 {
    padding: 20px 15px;
  }
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 1em;
}

.emotion-0 {
  margin: 6px 3px 6px 0;
  padding: 0px 4px;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1.54912em;
}

.emotion-2 {
  text-transform: uppercase;
  font-weight: 600;
  color: #fcba19;
}

.emotion-6 {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  color: #1A5A96;
  font-weight: 400;
  text-transform: capitalize;
  padding: 0 4px;
  margin: 0 2px;
}

.emotion-15 {
  width: 100%;
  max-width: 500px;
}

.emotion-11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 1.25em;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
}

.emotion-11 > input {
  margin: 0 4px;
  padding: 4px 5px;
  -webkit-flex-basis: 250px;
  -ms-flex-preferred-size: 250px;
  flex-basis: 250px;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-transition: all 0.25s;
  transition: all 0.25s;
}

.emotion-11 > button {
  padding: 4px 15px;
  margin: 4px;
}

.emotion-11 > button {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.emotion-8 {
  color: #fff;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  border: 1px solid transparent;
  padding: 0.375rem 0.75rem;
  font-size: 1em;
  line-height: 1.5;
  border-radius: 0.25rem;
  cursor: pointer;
  font-weight: 400;
  text-align: center;
  white-space: nowrap;
  vertical-align: middle;
  display: inline-block;
  background-color: #003366;
}

.emotion-13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
}

.emotion-17 {
  display: inline block;
  padding: 15px;
  max-width: 220px;
}

.emotion-17 img {
  width: 100%;
  margin-bottom: 0;
}

<div
  class="emotion-19 emotion-20"
>
  <div
    class="emotion-4 emotion-5"
  >
    <h2
      class="emotion-0 emotion-1"
    >
      DevHub
    </h2>
    <div
      aria-label="This application is currently in beta phase"
      class="emotion-2 emotion-3"
    >
      beta
    </div>
  </div>
  <h4
    css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
  >
    One place that brings together resources to help build digital products for the BC Government
  </h4>
  <a
    class="emotion-6 emotion-7"
    css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
    data-testid="ui-gatsby-link"
    href="/aboutDevhub"
  >
    How Devhub Works
     
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-chevron-right fa-w-10 "
      data-icon="chevron-right"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 320 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z"
        fill="currentColor"
      />
    </svg>
  </a>
  <div
    class="emotion-15 emotion-16"
  >
    <div
      class="emotion-10 emotion-11 emotion-12"
    >
      <input
        aria-label="Search input field"
        class="Input custom-text"
        data-testid="searchbar-input"
        id="dh-search-input"
        maxlength="40"
        type="text"
        value=""
      />
      <button
        aria-label="Click to Search"
        class="emotion-8 emotion-9"
        color="blue"
        data-testid="searchbar-button"
      >
        Search
      </button>
    </div>
    <div
      class="emotion-13 emotion-14"
    />
  </div>
  <div
    class="emotion-17 emotion-18"
    data-testid="Masthead.algolia"
    style="padding-bottom: 5px;"
  >
    <a
      href="https://www.algolia.com/"
    >
      <img
        alt="Algolia Logo"
        src="algolia.svg"
      />
    </a>
  </div>
</div>
`;
